src/sqlfluff/core/linter/linted_dir.py

```diff
@@ -115,19 +115,9 @@ def persist_changes(
         # Run all the fixes for all the files and return a dict
         buffer: Dict[str, Union[bool, str]] = {}
         for file in self.files:
-            if file.num_violations(fixable=True, **kwargs) > 0:
-                buffer[file.path] = file.persist_tree(suffix=fixed_file_suffix)
-                result: Union[bool, str]
-                if buffer[file.path] is True:
-                    result = "FIXED"
-                else:  # pragma: no cover
-                    result = buffer[file.path]
-            else:  # pragma: no cover TODO?
-                buffer[file.path] = True
-                result = "SKIP"
-
-            if formatter:
-                formatter.dispatch_persist_filename(filename=file.path, result=result)
```

Comment:
Most of this logic has been moved up into `.persist_tree`. It didn't seem to make much sense that it existed in the `LintedDir`, and that also made it harder to call sequentially during the linting step.